### PR TITLE
Start using error wrapping #792

### LINKED ISSUE #792: Start using error wrapping
# Expected Behavior

When we have layers of function calls that can each return errors, e.g. A->B->C where C returns an error, B needs to return it and then A needs to return that, we should use B and A to add context to that error, but retain the original error from C, e.g.:

```
_, err := ioutil.ReadAll(r)
if err != nil {
        return errors.Wrap(err, "read failed")
}
```

# Actual Behavior

Throughout our codebase, we have many instances of errors being "wrapped" via string formatting, e.g.:

https://github.com/tektoncd/pipeline/blob/11e03b73e549ab1508827182f6cc05ccf1a3080c/pkg/reconciler/v1alpha1/pipelinerun/resources/pipelinerunresolution.go#L109-L112

Instead of actually wrapping the error, we turn it into a string. And sometimes we can have 3+ levels of nesting within error messages like this!

# Additional Info

https://github.com/pkg/errors is a solid candidate; we should use this unless we have a strong reason to use something else.

A couple ways that completing this issue could look:

- Replace all error instances with wrapped errors (probably too tedious and time consuming to be feasible)
- (More likely) add the dependency + replace a few instances; we can use this as a precedent going forward

### PR DESCRIPTION
**Pull request by u5surf**
# Changes

fixes #792 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [-] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
Not functionality changed

- [-] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
Maybe not user facing

- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: pkg/reconciler/v1alpha1/pipelinerun/resources/pipelinerunresolution.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ import (
 	"time"
 
 	"github.com/knative/pkg/apis"
+	"github.com/okg/errors"
```
Comment (sbwsg) [inline pkg/reconciler/v1alpha1/pipelinerun/resources/pipelinerunresolution.go]: `okg` -> `pkg`

**Comment by ImJasonH** [comment]
I wonder if we should start using https://godoc.org/golang.org/x/xerrors -- this library will be moved into the errors package in the stdlib in a Go 1.13

**Comment by u5surf** [comment]
unit test:
https://tekton-releases.appspot.com/build/tekton-prow/pr-logs/pull/tektoncd_pipeline/808/pull-tekton-pipeline-unit-tests/1122889442651017217/
Result | FAILURE
Tests | 0 failed / 514 succeeded

What’s the meaning of Result `FAILURE` , although `Tests` was 0 failed?

**Comment by vdemeester** [comment]
@u5surf there is a few errors (`build-tests` and `unit-tests`) if you look at the logs

```
I0429 15:46:27.979] # github.com/tektoncd/pipeline/test/logs/pod
I0429 15:46:27.979] test/logs/pod/watcher.go:22:2: errors redeclared as imported package name
I0429 15:46:27.980] 	previous declaration at test/logs/pod/watcher.go:20:2
# […]
I0429 15:47:18.026] # github.com/tektoncd/pipeline/test/logs/pod
I0429 15:47:18.029] test/logs/pod/watcher.go:22:2: errors redeclared as imported package name
I0429 15:47:18.030] 	previous declaration at test/logs/pod/watcher.go:20:2
```

and 

```
I0429 15:51:44.055] Diffing /go/src/github.com/tektoncd/pipeline against freshly generated codegen
I0429 15:51:44.068] diff -Naupr /go/src/github.com/tektoncd/pipeline/pkg/client/informers/externalversions/generic.go /go/src/github.com/tektoncd/pipeline/tmpdiffroot.BL7fDM/pkg/client/informers/externalversions/generic.go
I0429 15:51:44.069] --- /go/src/github.com/tektoncd/pipeline/pkg/client/informers/externalversions/generic.go	2019-04-29 15:47:35.422100003 +0000
I0429 15:51:44.070] +++ /go/src/github.com/tektoncd/pipeline/tmpdiffroot.BL7fDM/pkg/client/informers/externalversions/generic.go	2019-04-29 15:45:02.163494373 +0000
I0429 15:51:44.070] @@ -16,8 +16,7 @@ limitations under the License.
I0429 15:51:44.070]  package externalversions
I0429 15:51:44.071]  
I0429 15:51:44.071]  import (
I0429 15:51:44.071] -	"fmt"
I0429 15:51:44.071] -
I0429 15:51:44.072] +	"github.com/pkg/errors"
I0429 15:51:44.072]  	v1alpha1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
I0429 15:51:44.073]  	schema "k8s.io/apimachinery/pkg/runtime/schema"
I0429 15:51:44.073]  	cache "k8s.io/client-go/tools/cache"
I0429 15:51:44.074] @@ -65,5 +64,5 @@ func (f *sharedInformerFactory) ForResou
I0429 15:51:44.074]  
I0429 15:51:44.074]  	}
I0429 15:51:44.075]  
I0429 15:51:44.075] -	return nil, fmt.Errorf("no informer found for %v", resource)
I0429 15:51:44.075] +	return nil, errors.Errorf("no informer found for %v", resource)
I0429 15:51:44.076]  }
I0429 15:51:45.407] /go/src/github.com/tektoncd/pipeline is out of date. Please run hack/update-codegen.sh
```

**Comment by ImJasonH** [comment]
Ping about the comment in https://github.com/tektoncd/pipeline/pull/808#issuecomment-487595982 -- have we considered using `xerrors` wrapping so that when it gets added to the stdlib in Go 1.13 we can easily move to that?

**Comment by u5surf** [comment]
@ImJasonH 
I have understood the usage of xerrors which can rewrite while keeping original format only changing “: %w”. Certainly it’s easier than rewriting with `pkg/errors`. I will retry this issue using it

File: test/wait.go [hunk 2]
```diff
@@ -48,6 +48,7 @@ import (
 	"fmt"
 	"time"
 
+  "golang.org/x/xerrors"
```
Comment (sbwsg) [inline test/wait.go]: Indent looks off here, 2 spaces -> 1 tab.

File: third_party/VENDOR-LICENSE [hunk 3]
```diff
@@ -1,8161 +0,0 @@
-
```
Comment (sbwsg) [inline third_party/VENDOR-LICENSE]: I'm curious - why did this file change as part of this PR?  Result of an automated process?

Comment (u5surf) [reply third_party/VENDOR-LICENSE]: It’s result of `./hack/update-codegen.sh`.

**Comment by vdemeester** [comment]
Related proposal for `golang/x/xerrors` : https://go.googlesource.com/proposal/+/master/design/29934-error-values.md

**Comment by vdemeester** [comment]
@u5surf can you rebase ? :angel:

**Comment by vdemeester** [comment]
oh noes, another rebase :sweat_smile: :bowing_man:

**Comment by bobcatfish** [comment]
> oh noes, another rebase

so close!!! sorry abotu this @u5surf we're almost there :pray:

**Pull request merged**